src/videojs.markers.js

```diff
@@ -1,9 +1,25 @@
+/*! videojs-markers - v0.4.0 - 2015-03-13
+* Copyright (c) 2015 ; Licensed  */
 /*! videojs-markers !*/
 'use strict'; 
 
 (function($, videojs, undefined) {
    //default setting
-   var defaultSetting = {
+   var defaultSetting = { 
```

Comment:
I see. That makes sense. The text aspect for your example can already be achieved with the markerTip::text function, in which you can use any attribute in the marker object. The time attribute can't be done yet. In terms of the code change, i would keep the text function and add another function time in markerTip which returns the time.

I don't see the need for 'setTime' and 'setText' since you can directly modify the data in the underlying marker object. That being said, we do need a function like 'markersUpdate' to let the plugin update the UI when the array of objects are modified. 

I'll write up a small code change and see if it'll fit your use case!